### ISSUE TITLE
Reload the notebook when the file changes on disk

### ISSUE THREAD

**Issue by ibash** [marimo-team/marimo#2675]
### Description

Neovim is my editor. When I edit a marimo notebook file and save it in vim, I would like marimo to reload with my changes.

Note that I am not editing external modules, but the marimo notebook directly.

It's not necessary to auto-run cells. I'm happy to run them manually.

Note that I think listening to file changes on disk is better than a direct neovim integration, because listening to file changes would allow integration with other tools.

Related to #998, however only module reloading was implemented.

### Suggested solution

Watch file changes to the notebook file on disk. When a change is detected update cells and mark which cells are outdated.

**Comment by gabrielgrant** [comment]
yes, big benefit of marimo being just python is that it easily integrates with other tooling and normal code editing environment. In addition to being nice to edit with a normal code editor, this also means that a notebook may sometimes be automatically edited by these other tools (eg when changing a symbol name in or moving an imported file). In these cases it seems especially important to pick up the changes, rather than just silently overwrite (which seems to be the default behavior today?)

**Comment by gaardhus** [comment]
Related to #1511

**Comment by gabrielgrant** [comment]
@dmadisetti you mention that the VSCode plugin does this. Can you explain a bit more how that works? I'd love to be able to just edit the .py file in VSCode and have a kiosk mode updated with the granular (cell-level) execution. this could be automatically on save, or the same way it works when re-running in the notebook interface where save and run are separate (ie cell outputs can be "dirty")

**Comment by dmadisetti** [comment]
@gabrielgrant That should work today?

The VSCode plugin is very similar to the jupyter notebook experience

Mind opening an issue in https://github.com/marimo-team/vscode-marimo/issues if the problem persists?

**Comment by gabrielgrant** [comment]
I haven't been able to use the VScode plugin notebook experience (seem to be having the same issue as https://github.com/marimo-team/vscode-marimo/issues/61 ), so what I was talking about was just editing the .py file directly. Are you saying that is expected to stay in sync with what I'm seeing in the browser? It doesn't seem to.

**Comment by dmadisetti** [comment]
Hmm, it may be that the plugin is currently broken. What is supposed to happen is very similar to the jupyter experience (none of the generated `_` functions should be present, just cells).

Just editing a file will not keep it in sync, only through the plugin interface that seems to currently have an issue.

**Comment by sneerin** [comment]
the cursor experience is totally broken,  review as code -> notebook is not refreshed -> you close nb with messge don't save changes -> open it agin. even old jupyter nb is usable but not marimo

**Comment by gabrielgrant** [comment]
@sneerin are you using the `--watch` flag added in the PR that closed this issue?

Details: https://docs.marimo.io/guides/editor_features/watching/
